ComponentKit/Debug/CKComponentDebugController.mm

```diff
@@ -133,46 +132,35 @@ + (BOOL)debugMode
 
 #pragma mark - Synchronous Reflow
 
+static NSHashTable<id<CKComponentDebugReflowListener>> *reflowListeners;
+static std::mutex reflowMutex;
+
 + (void)reflowComponents
 {
   if (![NSThread isMainThread]) {
     dispatch_async(dispatch_get_main_queue(), ^{ [self reflowComponents]; });
-  } else {
-    UIWindow *window = [[UIApplication sharedApplication] keyWindow];
-    CKRecursiveComponentReflow(window);
+    return;
   }
-}
-
-+ (void)reflowComponentsForView:(UIView *)view searchUpwards:(BOOL)upwards
-{
-  if (upwards) {
-    while (view && ![view isKindOfClass:[CKComponentRootView class]]) {
-      view = view.superview;
-    }
+  NSArray<id<CKComponentDebugReflowListener>> *copiedListeners;
+  {
+    std::lock_guard<std::mutex> l(reflowMutex);
+    copiedListeners = [reflowListeners allObjects];
   }
-  if (view) {
-    CKRecursiveComponentReflow(view);
+  for (id<CKComponentDebugReflowListener> listener in copiedListeners) {
+    [listener didReceiveReflowComponentsRequest];
```

Comment:
This is so much better than traversing the view hierarchy to find the component lifecycle manager or explicitly telling component hosting views to layout.